src/main/java/com/mraof/minestuck/item/StructureScannerItem.java

```diff
@@ -1,85 +1,132 @@
 package com.mraof.minestuck.item;
 
-import net.minecraft.ChatFormatting;
+import com.mraof.minestuck.Minestuck;
+import com.mraof.minestuck.client.ClientProxy;
+import net.minecraft.client.gui.Gui;
+import net.minecraft.client.renderer.item.CompassItemPropertyFunction;
+import net.minecraft.client.renderer.item.ItemProperties;
 import net.minecraft.core.BlockPos;
-import net.minecraft.network.chat.Component;
+import net.minecraft.core.GlobalPos;
+import net.minecraft.nbt.NbtUtils;
+import net.minecraft.resources.ResourceLocation;
 import net.minecraft.server.level.ServerLevel;
+import net.minecraft.server.level.ServerPlayer;
 import net.minecraft.sounds.SoundEvents;
 import net.minecraft.sounds.SoundSource;
 import net.minecraft.tags.TagKey;
 import net.minecraft.world.InteractionHand;
 import net.minecraft.world.InteractionResultHolder;
+import net.minecraft.world.entity.Entity;
 import net.minecraft.world.entity.player.Player;
 import net.minecraft.world.item.Item;
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.level.Level;
 import net.minecraft.world.level.levelgen.structure.Structure;
 
+import javax.annotation.Nullable;
 import java.util.function.Supplier;
 
 public class StructureScannerItem extends Item
 {
 	private final TagKey<Structure> structure;
+	@Nullable
 	private final Supplier<Item> fuelItem;
 	
-	public StructureScannerItem(Properties properties, TagKey<Structure> structure, Supplier<Item> fuelItem)
+	public StructureScannerItem(Properties properties, TagKey<Structure> structure, @Nullable Supplier<Item> fuelItem)
 	{
 		super(properties);
 		this.structure = structure;
 		this.fuelItem = fuelItem;
+		
+		ItemProperties.register(this, new ResourceLocation("angle"),
+				new CompassItemPropertyFunction((level, stack, entity) -> {
+					if(stack.hasTag() && stack.getTag().contains("TargetLocation"))
+					{
+						return GlobalPos.of(level.dimension(), NbtUtils.readBlockPos(stack.getTag().getCompound("TargetLocation")));
+					} else
+					{
+						return null;
+					}
+				}));
+		
+		ItemProperties.register(this, new ResourceLocation(Minestuck.MOD_ID, "powered"),
+				((pStack, pLevel, pEntity, pSeed) -> pStack.hasTag() && pStack.getTag().getBoolean("Powered") ? 1 : 0));
 	}
 	
 	@Override
-	public InteractionResultHolder<ItemStack> use(Level levelIn, Player playerIn, InteractionHand handIn)
+	public InteractionResultHolder<ItemStack> use(Level pLevel, Player pPlayer, InteractionHand pUsedHand)
 	{
-		levelIn.playSound(playerIn, playerIn.getX(), playerIn.getY(), playerIn.getZ(), SoundEvents.UI_BUTTON_CLICK.value(), SoundSource.AMBIENT, 0.8F, 1.3F);
+		ItemStack stack = pPlayer.getItemInHand(pUsedHand);
 		
-		ItemStack item = playerIn.getItemInHand(handIn);
-		boolean foundItem = false;
+		if(pPlayer.isCreative() | stack.getDamageValue() != stack.getMaxDamage() | checkFuelItem(pPlayer, pLevel))
```

Comment:
I suspect that you want the conditional OR instead of the logical OR here due to the side effects of `checkFuelItem()`. Is that right, or is the logical OR intentional here?

Reply:
Ah, is that the difference? I was wondering why it allowed me to use a single symbol. I'll resolve that right away.